ann_cnn_training/model_definition.py

```diff
@@ -361,3 +361,132 @@ def totalsize(self):
+            self.conv1 = nn.Conv2d(
+                in_channels=idim, out_channels=idim, kernel_size=self.stencil, stride=1, padding=0
+            )
+            self.act_cnn = nn.ReLU()
+            self.dropout0 = nn.Dropout(p=0.5 * self.dropout_prob)
+
+        # can define a block and divide it into blocks as well
+        self.layer1 = nn.Linear(idim, hdim)  # ,dtype=torch.float16)
+        self.act1 = nn.LeakyReLU()
+        # self.bnorm1 = nn.BatchNorm1d(hdim)
+
+        self.dropout = nn.Dropout(p=self.dropout_prob)
+
+        self.layer2 = nn.Linear(hdim, hdim)
+        self.act2 = nn.LeakyReLU()
+        # self.bnorm2 = nn.BatchNorm1d(hdim)
+        # -------------------------------------------------------
+        self.layer3 = nn.Linear(hdim, hdim)
+        self.act3 = nn.LeakyReLU()
+        # self.bnorm3 = nn.BatchNorm1d(hdim)
+        # -------------------------------------------------------
+        self.layer4 = nn.Linear(hdim, hdim)
+        self.act4 = nn.LeakyReLU()
+        # self.bnorm4 = nn.BatchNorm1d(2 * hdim)
+        # --------------------------------------------------------
+        self.layer5 = nn.Linear(hdim, hdim)
+        self.act5 = nn.LeakyReLU()
+        # self.bnorm5 = nn.BatchNorm1d(hdim)
+        # -------------------------------------------------------
+
+        self.layer6 = nn.Linear(hdim, hdim)
+        self.act6 = nn.LeakyReLU()
+        # -------------------------------------------------------
+        self.layer7 = nn.Linear(hdim, hdim)
+        self.act7 = nn.LeakyReLU()
+        # -------------------------------------------------------
+        self.layer8 = nn.Linear(hdim, hdim)
+        self.act8 = nn.LeakyReLU()
+        # -------------------------------------------------------
+        self.layer9 = nn.Linear(hdim, hdim)
+        self.act9 = nn.LeakyReLU()
+        # -------------------------------------------------------
+        self.layer10 = nn.Linear(hdim, 2 * odim)
+        self.act10 = nn.LeakyReLU()
+        # -------------------------------------------------------
+
+        self.output = nn.Linear(2 * odim, odim)
+
+    def forward(self, x):
+        if self.fac == 1:
+            x = torch.squeeze(self.dropout0(self.act_cnn(self.conv1(x))))
+        elif self.fac == 2:
+            x = torch.squeeze(self.dropout0(self.act_cnn(self.conv1(x))))
+            x = torch.squeeze(self.dropout0_2(self.act_cnn2(self.conv2(x))))
+
+        x = self.dropout(self.act1(self.layer1(x)))
+        x = self.dropout(self.act2(self.layer2(x)))
+        x = self.dropout(self.act3(self.layer3(x)))
+        x = self.dropout(self.act4(self.layer4(x)))
+        x = self.dropout(self.act5(self.layer5(x)))
+        x = self.dropout(self.act6(self.layer6(x)))
+        x = self.dropout(self.act7(self.layer7(x)))
+        x = self.dropout(self.act8(self.layer8(x)))
+        x = self.dropout(self.act9(self.layer9(x)))
+        x = self.dropout(self.act10(self.layer10(x)))
```

Comment:
similarly here:

We should gather this into a `list` and loop over the layers and activation functions